core-tests/jvm/src/test/scala/zio/RTSSpec.scala

```diff
@@ -62,10 +62,12 @@ object RTSSpec extends ZIOBaseSpec {
       for {
         promise <- Promise.make[Nothing, Int]
         fiber   <- promise.await.fork
-        dump    <- fiber.dump
-        dumpStr <- dump.fold[URIO[Clock, String]](IO.succeed(""))(_.prettyPrintM)
-        _       <- UIO(println(dumpStr))
-      } yield assert(dump)(anything)
+        dumpStr <- fiber.typeSpecific(
```

Comment:
When you call `fork` on a fiber, you know it's a runtime fiber. So we can change the signature of `fork*` methods to return a `Fiber.Runtime`. Then this code will be simple again.

Reply:
We should look for other places where we know it's a real fiber, and change them to be `Fiber.Runtime`.

Reply:
Yes. I think we may need to move a couple more implementations down from `Fiber` to `Runtime` and `Synthetic`. For example mapping a fiber would return the same type as the original fiber I believe.

Reply:
Well, due to covariance, we can override `map` to preserve the subtype of the parent (e.g. in `Runtime#map`, it returns a `Runtime`, but in `Synthetic#map`, it returns a `Synthetic`). But is that actually legit? Technically, a mapped fiber is a synthetic fiber that happens to be 1-to-1 with the original. I think we should preserve the invariant that, essentially, `FiberContext` is the only real `Runtime` fiber, and all the others are `Synthetic`. 

We could always add a method like `childrenRuntime` to grab the children and filter to those that are `Runtime` if that proves helpful.. then if you map and grab the childrenRuntime, you can find the original one again

Reply:
Yes, I think that is fair. I was thinking of something like `mapM` where `id` and `trace` currently return the same values as the original fibers but maybe that is not legitimate and being more explicit about this is actually forcing us to clean it up.